types/eslint-utils/referenceTracker.d.ts

```diff
@@ -0,0 +1,61 @@
+import { Scope } from 'eslint';
+import { Node } from 'estree';
+
+export interface ReferenceTrackerOptions {
+    /**
+     * The variable names for Global Object.
+     */
+    globalObjectNames?: string[];
+
+    /**
+     * The mode to determine the ImportDeclaration's behavior for CJS modules.
+     */
+    mode?: 'legacy' | 'strict';
+}
+
+export interface TraceMap {
+    [i: string]: TraceMapObject;
+}
+
+export interface TraceMapObject {
+    [CALL]?: boolean;
+    [CONSTRUCT]?: boolean;
+    [ESM]?: boolean;
+    [READ]?: boolean;
+}
```

Comment:
TraceMap can have a deep hierarchy.

```suggestion
export interface TraceMapObject {
    [i: string]: TraceMapObject
    [CALL]?: boolean;
    [CONSTRUCT]?: boolean;
    [ESM]?: boolean;
    [READ]?: boolean;
}
```

https://eslint-utils.mysticatea.dev/api/scope-utils.html#tracker-iterateglobalreferences